app/models/spree/product.rb

```diff
@@ -455,7 +455,7 @@ def add_primary_taxon_to_taxons
     end
 
     def remove_previous_primary_taxon_from_taxons
-      return unless primary_taxon_id_changed? && primary_taxon_id_was
+      return unless saved_change_to_primary_taxon_id? && primary_taxon_id_was
```

Comment:
Is this `#primary_taxon_id_was` still okay..? :thinking:

Reply:
I notice there's still a lot of failing model specs for Product related to the taxon....

Reply:
Nice spot, 43a38cdaa4cb110995bc01ea2807a35d16263261